### ISSUE TITLE
Permission for seeing expenses

### ISSUE THREAD

**Issue by jupsoleil** [dhanasingh/redmine_wktime#44]
Hello, let me start by saying this is a fantastic plugin!!

I don't find a way to limit wk_expense_entries/reportdetail to see only the user's own expense declarations, so only a certain group of managers can oversee all declared expenses.  Similar to 'edit time logs' and 'edit own time logs'.  
Am I doing something wrong, or is this feature currently not available?

Kind regards,

Jürgen

**Comment by suganya-thulasiraman** [comment]
Yes, currently if user has 'View spent time' permission, then he can see all expense entries. But if we want to build this, we can implement like this, if the user has 'Edit time log' permission then he can view other users expenses. 
We could probably include this in furure release.

**Comment by jupsoleil** [comment]
I would rather propose to add permissions specific to expenses to make it more general, but this will probably create more work for the developer.

**Comment by jupsoleil** [comment]
I've temporarily modified redmine/plugins/redmine_wktime/init.rbinit.rb to be as follows (last line; permission approve_time_entries instead of edit_time_entries):
```
...
# Added expense report link in redmine core 'projects/show.html' using hook
	def view_projects_show_sidebar_bottom(context={})
		if !context[:project].blank?
			wktime_helper = Object.new.extend(WktimeHelper)		
			host_with_subdir = wktime_helper.getHostAndDir(context[:request])	
			project_ids = Setting.plugin_redmine_wktime['wkexpense_projects']		
			if project_ids.blank? || (!project_ids.blank? && (project_ids == [""] || project_ids.include?("#{context[:project].id}"))) && User.current.allowed_to?(:approve_time_entries, context[:project])
...
```
It's not perfect, but serves my purpose for the time being.

**Comment by karthickmadhiazhagan** [comment]
We have fixed this in v2.2.1. Only TE admin or the user who have 'Edit time log' permission can view other users expenses.
